File: kotlin/canSum.kt
fun canSum(targetSum: Int, numbers: Array<Int>): Boolean {
    if (targetSum < 0) return false;
    if (targetSum == 0) return true;

    numbers.forEach {
        if (canSum(targetSum - it, numbers)) return true
    }

    return false
}

fun main(vararg args: String): Unit {
    println(canSum(args[0].toInt(), arrayOf<Int>(14, 7)))
}

File: javascript/canSum.js
const canSum = (target, numbers) => {
    if (target < 0) return false;
    if (target === 0) return true;

    for (let i of numbers)
        if (canSum(target - i, numbers)) return true

    return false
}

console.log(canSum(3000, [7, 14]))